File: LeaveManegamentProject/LeaveManegamentProject/Controllers/LeaveController.cs
using LeaveManegamentProject.DbContexts;
using LeaveManegamentProject.Entities;
using LeaveManegamentProject.Helper;
using LeaveManegamentProject.ViewModels;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Mail;
using System.Web;
using System.Web.Mvc;

namespace LeaveManegamentProject.Controllers
{
    public class LeaveController : Controller
    {
        LeaveContext context = new LeaveContext();
        Response response = new Response();

        [HttpGet]
        public string Index()
        {

            
            var leavelist = context.Leaves.ToList();
            return JsonConvert.SerializeObject(AutoMapper.Mapper.Map<LeaveView[]>(leavelist));

        }
        string ManagerEmail = string.Empty;
        string Email = string.Empty;
        [HttpPost]
        public string emailsend(Emailclass ec)
        {
            var adminlist = context.Admins.ToList();
            foreach (var admin in adminlist){
                if (ec.ProjectName == admin.ProjectName)
                {
                    if (admin.Designation == "Manager")
                    {
                        ManagerEmail = admin.Email;
                    }
                    if(admin.Role=="Team Lead")
                    {
                        Console.WriteLine(admin.Email);
                       // Email= admin.Email;
                    }
                }

            }
            string To = ManagerEmail;
            string Cc = ec.From;
            string body = ec.Body;
            string subject = ec.Subject;
            string from = ec.From;
            string Password = ec.Password;


            MailMessage mm = new MailMessage();
            mm.From = new MailAddress(from);
          
            mm.Subject = subject;
            mm.Body = body;
           // mm.Cc = CC;


            mm.To.Add(To);
            mm.To.Add(Cc);
            mm.IsBodyHtml = false;

            SmtpClient smtp = new SmtpClient("smtp.gmail.com");
            smtp.Port = 587;
            smtp.UseDefaultCredentials = true;
            smtp.EnableSsl = true;
            smtp.Credentials = new System.Net.NetworkCredential(ec.From, ec.Password);
            smtp.Send(mm);
            return "The Mail Has Been Sent To" + To.ToString();
        }

        [HttpPost]
        public string Save(LeaveView leaveview)
        {
            var Leave = AutoMapper.Mapper.Map<Leave>(leaveview);

            if (string.IsNullOrEmpty(leaveview.AbsenceType))
            {
                response.Message = "please Provide Absence Name";
                response.IsSuccess = false;
                return JsonConvert.SerializeObject(response);
            }
            if ((leaveview.Days) == 0)
            {
                response.Message = "Calculate Dates";
                response.IsSuccess = false;
                return JsonConvert.SerializeObject(response);
            }



            context.Leaves.Add(Leave);
            context.SaveChanges();
            response.Message = "Record Save sucessfully!";
            response.IsSuccess = true;

            return JsonConvert.SerializeObject(response);
        }





        [HttpGet]
        public string Delete(Int64 id)
        {
            var leave = context.Leaves.Find(id);

            if (leave != null)
            {
                context.Leaves.Remove(leave);
                context.SaveChanges();
                response.IsSuccess = true;
                response.Message = "Record deleted successfully!";
            }

            return JsonConvert.SerializeObject(response);
        }


        [HttpGet]

        public string accept(Int64 id)
        {
            var leave = context.Leaves.Find(id);
            

            if (leave != null )
            {
                leave.Action = "Accept";
                leave.CurrentBalance = leave.CurrentBalance - leave.Days;
                //if (leave.CurrentBalance == 0)
                //{
                //    leave.CurrentBalance = 25 - leave.Days;
                //}
                //else
                //{
                //    leave.CurrentBalance = leave.CurrentBalance - leave.Days;
                //}


                context.Entry(leave).State = System.Data.Entity.EntityState.Modified;
                context.SaveChanges();
                response.IsSuccess = true;
                response.Data = leave;
                response.Message = "Record Accept successfully!";
            }

            return JsonConvert.SerializeObject(response);
        }
        [HttpGet]
        public string reject(Int64 id)
        {
            var leave = context.Leaves.Find(id);

            if (leave != null)
            {
                leave.Action = "Reject";

                context.Entry(leave).State = System.Data.Entity.EntityState.Modified;
                context.SaveChanges();
                response.IsSuccess = true;
                response.Message = "Record Accept successfully!";
            }

            return JsonConvert.SerializeObject(response);
        }
        [HttpPost]
        public string Reject(Int64 Id)
        {
            var leave = context.Leaves.Find(Email);
            return JsonConvert.SerializeObject(AutoMapper.Mapper.Map<LeaveView>(leave));
        }
        
      
    }
}

File: LeaveManegamentProject/LeaveManegamentProject/AppScripts/Admin/LeaveController.js


app.controller("LeaveController", ["$scope", "$http", "$stateParams", "$rootScope", "$location", "leaveservice",
    function ($scope, $http, $stateParams,  $rootScope,$location, leaveservice, ) {
       
        $scope.Leaves = [];
        $scope.IsVisible = false;
        $scope.GetAllData = function () {
            $http.get('/Leave').success(function (response) {
                console.log(response);
                $scope.Leaves = response;
                
            });
        };
        $scope.GetAllData();
        $scope.Delete = function (id) {

            $http.get("/Leave/Delete/" + id).success(function (response) {

                if (response.IsSuccess) {
                    leaveservice.RecordStatusMessage("success", response.Message);
                    $scope.GetAllData();
                }
                else {
                    leaveservice.RecordStatusMessage("danger", response.Message);
                }
            });
        };
        $scope.accept = function (Id) {

            $http.get("/Leave/accept/" + Id).success(function (response) {

                leaveservice.RecordStatusMessage("success", response.Message);
                console.log(response);
                $scope.Leave = response.Data;
                console.log($scope.Leave.Action);
                $scope.GetAllData();

                console.log($scope.Leave.CurrentBalance);



               

                $http.post("/Admin/date", { adminview: $scope.Leave.CurrentBalance, Email: $scope.Leave.Email }).success(function (response) {
                        console.log(response);
                        if (response.IsSuccess) {
                            $scope.Admin = {};
                            leaveservice.RecordStatusMessage("success", response.Message);
                            

                        } else {
                            leaveservice.RecordStatusMessage("error", response.Message);
                        }
                    });             
              
            });           
        }
        
                
        $scope.reject = function (id) {
            $http.get("/Leave/reject/" + id).success(function (response) {
                leaveservice.RecordStatusMessage("success", response.Message);    
                $scope.GetAllData();               
            });
        }
        if ($stateParams.email != undefined) {

            $scope.email = $stateParams.email;
            console.log($scope.ProjectName);
            return ($scope.email);
        }
           

    }]);

app.controller("LeaveAddController", ["$scope", "$http", "$stateParams", "$location", "leaveservice",
    function ($scope, $http,$stateParams, $location, leaveservice) {
       
        $scope.Leave = {};
       
      
           
           
              
     
        
        
        $scope.IsVisible = false;
     
        $scope.GetAllData = function () {
            $http.get('/Absencename').success(function (response) {
                console.log(response);
                $scope.Absencenames = response;
                
            });
        };
        $scope.GetAllData();

        $scope.IsEditMode = false;
       
        if ($stateParams.email != undefined) {
            $scope.IsEditMode = true;
             
            $http.post("/admin/edit", { email: $stateParams.email }).success(function (response) {
               
                console.log(response);
                console.log($stateParams);
             
                $scope.Leave = response;
                //$scope.IsVisible = "Admin";
                //$scope.Visible = "Team Lead";
                console.log($scope.IsVisible);
                if ($scope.Leave.Role == "Admin") {
                    $scope.IsVisible = $scope.IsVisible = true;

                }
                else if ($scope.Leave.Role == "Team Lead") {
                    $scope.Visible = $scope.Visible = true;
                }
                else {
                    $scope.IsVisible = $scope.IsVisible = false;
                    $scope.Visible = $scope.Visible = false;
                }
                //$scope.Leave.Date1 = new Date();
                $scope.date1 = new Date();
                $scope.ms = new Date().getTime() + 86400000*30;
                $scope.date2 = new Date($scope.ms);
               
                //if ($scope.date2 = new Date().getMonth() + 2 > 12) {
                //    $scope.date2 = new Date().getMonth() + 2 - 12;
                //}
                //else {
                //    $scope.date2 = new Date().getMonth() + 2;
                //}
                
                

            });        
        }

        $scope.calculate = function () {

            $scope.Leave.Action = "Pending";

            console.log($scope.Leave.Date1);
            console.log($scope.Leave.Date2);
            $scope.Day = $scope.Leave.Date1 - $scope.Leave.Date2;
            $scope.ab = (24 * 60 * 60 * 1000);
            
            if ($scope.Leave.Days = -($scope.Day / $scope.ab) + 1 <1) {
               
                alert("Check your Dates");
            }
             else  {
               
                $scope.Leave.Days = -($scope.Day / $scope.ab) + 1;
                
            }
           
        }
        $scope.GetAllData = function () {
            $http.get('/Admin').success(function (response) {
                console.log(response);
                $scope.Admins = response;

            });
        };
        $scope.GetAllData();
        $scope.sendmail = function () {
            
           
            console.log($scope.Leave.email);
            var butClick = document.getElementById("ButClick").getAttribute("value");
            console.log($scope.Leave.ProjectName);
            if (butClick == "Send") {
                $scope.ec = {};
                $scope.ec.ProjectName = $scope.Leave.ProjectName;
                $scope.ec.From = $scope.Leave.Email;;
                //$scope.ec.To = "kilupatil@gmail.com";
                $scope.ec.Body = "https://localhost:44336/#/leave";
                $scope.ec.Subject = "Leave Application";
                $scope.ec.Password = $scope.Leave.Password;
                $http({
                    method: 'POST',
                    url: '/Leave/emailsend',
                    data: JSON.stringify($scope.ec, $scope.ProjectName)
                }).then(function (result) {
                    alert(result.data);
                });
                $scope.Save();
            }
        }

        $scope.Save = function () {
          
          
            $http.post("/Leave/Save", { leaveview: $scope.Leave }).success(function (response) {
                console.log(response);
                if (response.IsSuccess) {
                    $scope.Leave = {};
                    leaveservice.RecordStatusMessage("success", response.Message);
                    

                } else {
                    leaveservice.RecordStatusMessage("error", response.Message);
                }
                  if ($stateParams.email != undefined) {
                    $scope.IsEditMode = true;

                    $http.post("/Admin/edit", { email: $stateParams.email }).success(function (response) {
                        console.log(response);
                        console.log($stateParams);

                        $scope.Leave = response;

                    });

                }
            });
              
           
           

        };

      

    }]);































